notebooks/hello-universe/Interpreting_CNNs/Interpreting_CNNs.ipynb

```diff
@@ -0,0 +1,803 @@
+    "    e2 = Dropout(0.5)(d2)\n",
+    "\n",
+    "    f = Flatten()(e2)\n",
+    "    z0 = Dense(64, activation='softmax', kernel_regularizer=l2(0.0001))(f)\n",
+    "    z1 = Dense(32, activation='softmax', kernel_regularizer=l2(0.0001))(z0)\n",
+    "    y_out = Dense(1, activation='sigmoid')(z1)\n",
+    "\n",
+    "    cnn = Model(inputs=x_in, outputs=y_out)\n",
+    "    return cnn"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d419c4cc-ad09-49db-b903-636eec3577d0",
+   "metadata": {},
+   "source": [
+    "**Since ablation analysis requires training multiple models**, it can often be more resource intensive than other methods. However, it can also provide useful information on the way features are organized during training. The following cell can be edited to change how much we want to train our mini-models."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ef61819e-4804-4c17-be05-31255e0522da",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# You can change how much to train each model\n",
+    "# 5 epochs is chosen due to time and computation constraints\n",
+    "num_epochs = 5\n",
+    "\n",
+    "# Train the original model\n",
+    "model = create_model()\n",
+    "model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])\n",
+    "model.fit(X_train, y_train, epochs=num_epochs, batch_size=32, validation_data=(X_test, y_test))\n",
+    "\n",
+    "# Get baseline performance\n",
+    "baseline_score = model.evaluate(X_test, y_test)\n",
+    "\n",
+    "# Ablate each layer and compare performance\n",
+    "layers_to_ablate = ['c0', 'c1', 'c2']\n",
+    "for layer in layers_to_ablate:\n",
+    "    model_ablated = create_model(ablate=layer)\n",
+    "    model_ablated.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])\n",
+    "    model_ablated.fit(X_train, y_train, epochs=num_epochs, batch_size=32, validation_data=(X_test, y_test))\n",
+    "    ablated_score = model_ablated.evaluate(X_test, y_test)\n",
+    "    \n",
+    "    print(f\"Performance drop after ablating {layer}: {baseline_score[1] - ablated_score[1]}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3e0a2ddf-665a-4ddc-a69b-417f149253b6",
+   "metadata": {},
+   "source": [
+    "## FAQ\n",
+    "\n",
+    "- **How do I choose which saliency model to use?** Each saliency technique has it's own strengths and weaknesses. If you are looking for an intuitive approach that produces the highest resolution image, then gradient backpropagation is the best choice. If you want to control which layer to look at and watch how the model evolves or are concerned about reliability, then Grad-CAM will perform better. If you don't have access to the inner workings of the model or want to understand how a model would perform with slightly different inputs, then RISE is the best option.\n",
```

Comment:
This doesn't mention when you'd want to use ablation analysis.  Is that intentional?  If I just read this knowing nothing but this notebook about these techniques (which is basically true for me...) I might conclude there's no use to ever doing ablation analysis.

Reply:
@oliverlin23 please provide a citation to ablation analysis for interpretability, along with a short sentence describing when it could be useful

Reply:
resolved via Slack conversation, associated edits incorporated.